properties/base.py

```diff
@@ -278,14 +281,20 @@ def assert_valid(self, instance, value=None):
             value.validate()
         return True
 
-    @staticmethod
-    def as_json(value):
+    def as_json(self, value):
```

Comment:
Can you clarify what you want here? It's not clear to me that we necessarily want to be overriding the pickle behaviour if we are setting a JSON serializer...?